### PR TITLE
Vlan filter based off the filter engine

### PR DESCRIPTION
**Pull request by jrouzierinverse**
## Description

Vlan filters based off the filter engine depends on feature/filter-engine
## NEWS file entries

New Features
++++++++++++

Enhancements
++++++++++++

Bug Fixes
+++++++++
## Delete branch after merge

NO

### CODE REVIEW

**Comment by jrouzierinverse** [comment]
This depends on the filter engine

File: lib/pf/factory/condition/vlanfilter.pm [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+our @MODULES;
+
+sub factory_for {'pf::condition'}
+
+our %VLAN_FILTER_TYPE_TO_CONDITION_TYPE = (
+    'is'        => 'pf::condition::equals',
+    'is_not'    => 'pf::condition::not_equals',
+    'match'     => 'pf::condition::matches',
+    'match_not' => 'pf::condition::not_matches',
```
Comment (fdurand) [inline lib/pf/factory/condition/vlanfilter.pm]: if we can have a defined condition

File: lib/pf/factory/condition/vlanfilter.pm [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+);
+
+our %VLAN_FILTER_KEY_TYPES = (
```
Comment (fdurand) [inline lib/pf/factory/condition/vlanfilter.pm]: ssid, connection_type, username ... ?

Comment (rouzier) [inline lib/pf/factory/condition/vlanfilter.pm]: This is for types that are hashes not simple values.
{
node_info => {...},
ssid => 'BigBob',
connection_type => 'pigeon'
}

**Comment by jrouzierinverse** [comment]
Please Review

File: addons/dev-helpers/dump.pl [hunk 3]
```diff
@@ -70,14 +70,14 @@ sub _run {
...
 }
 
-package pf::dump::vlan_filters;
+package pf::dump::access_filters;
 use base qw(pf::dump::cmd);
 use Data::Dumper;
 __PACKAGE__->mark_as_loaded();
 
 sub _run {
-    require pf::vlan::filter;
```
Comment (fdurand) [inline addons/dev-helpers/dump.pl:79]: Why don't you keep pf::vlan::filter ? I like the way to have that explicit and use the same logic for radius per example (pf::radius::filter)

**Pull request closed without merging**